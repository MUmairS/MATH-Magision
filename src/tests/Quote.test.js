import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Quote from '../components/Quote';

describe('renders correctly', () => {
  test('render home-page', () => {
    const quote = render(<Quote />);
    expect(quote).toMatchSnapshot();
  });

  test('check title exist', () => {
    render(<Quote />);
    expect(
      screen.getByText(/Math is the source code of reality./i),
    ).toBeInTheDocument();
  });
});
