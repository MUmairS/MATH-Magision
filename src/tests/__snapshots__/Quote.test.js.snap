// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly render home-page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="quoteContainer"
      >
        <div
          class="quotes"
        >
          <p>
            "Math is the source code of reality."
          </p>
        </div>
        <p
          class="quotes"
        />
        <button
          class="generateQuotes"
          type="button"
        >
          Generate quote
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="quoteContainer"
    >
      <div
        class="quotes"
      >
        <p>
          "Math is the source code of reality."
        </p>
      </div>
      <p
        class="quotes"
      />
      <button
        class="generateQuotes"
        type="button"
      >
        Generate quote
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
